File: sdk/php/src/Subjects.php
<?php

declare(strict_types=1);

namespace Cordum\Cap;

/**
 * NATS subject constants and protocol defaults for the Cordum Agent Protocol.
 */
final class Subjects
{
    public const SUBMIT = 'sys.job.submit';
    public const RESULT = 'sys.job.result';
    public const HEARTBEAT = 'sys.heartbeat';
    public const ALERT = 'sys.alert';
    public const PROGRESS = 'sys.job.progress';
    public const CANCEL = 'sys.job.cancel';
    public const DLQ = 'sys.job.dlq';
    public const WORKFLOW_EVENT = 'sys.workflow.event';
    public const HANDSHAKE = 'sys.handshake';

    public const DEFAULT_PROTOCOL_VERSION = 1;
    public const DEFAULT_HEARTBEAT_INTERVAL = 5; // seconds
}

File: sdk/rust/src/subjects.rs
use std::time::Duration;

pub const SUBJECT_SUBMIT: &str = "sys.job.submit";
pub const SUBJECT_RESULT: &str = "sys.job.result";
pub const SUBJECT_HEARTBEAT: &str = "sys.heartbeat";
pub const SUBJECT_ALERT: &str = "sys.alert";
pub const SUBJECT_PROGRESS: &str = "sys.job.progress";
pub const SUBJECT_CANCEL: &str = "sys.job.cancel";
pub const SUBJECT_DLQ: &str = "sys.job.dlq";
pub const SUBJECT_WORKFLOW_EVENT: &str = "sys.workflow.event";
pub const SUBJECT_HANDSHAKE: &str = "sys.handshake";

pub const DEFAULT_PROTOCOL_VERSION: i32 = 1;
pub const DEFAULT_HEARTBEAT_INTERVAL: Duration = Duration::from_secs(5);
